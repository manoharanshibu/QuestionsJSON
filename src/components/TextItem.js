import React, { Component } from 'react';

export default class TextItem extends Component{
    
    createItems(nodes, states, childIndex = 0) {
        if(!nodes && !states) return;
        let title;
        switch(childIndex) {
            case 0:
                title = `<h1>${nodes.title}</h1>`;
                break;
            case 1:
                title = `<h2>${nodes.title}</h2>`;
                break;
            case 2:
                title = `<h3>${nodes.title}</h3>`;
                break;
            default:
                title = `<h4>${nodes.title}</h4>`;
        }

        let parentNode = `<div class="node">${title}</div>`;
        if (nodes.qas) {
            let qas = `<ul class="qas">`
            nodes.qas.map((item, index) => {
                qas += `<li class="liclass ${(states.indexOf(item.tocId) !== -1 ? ' selected' : ' ')}"> <div class="quest">Q${index+1}. ${item.question}</div> <div class="ans">${item.answer}</div>`;
            });
            qas += `</ul>`;
            parentNode = `<div class="node">${title} ${qas}</div>`;
        }

        if (nodes.children) {
            nodes.children.map(item => {
                parentNode += this.createItems(item, states, ++childIndex);
            });
        }

        return parentNode;
    }

    render(){
        const { nodes, states } = this.props
        return <div class="node"><div className='tree-container' dangerouslySetInnerHTML={{__html: this.createItems(nodes, states, 0)}} ></div></div>
    }

}