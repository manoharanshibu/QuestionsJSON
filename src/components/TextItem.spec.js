import React from 'react';
import renderer from 'react-test-renderer';

import TextItem from './TextItem';

describe('snap shot test', () => {
  it('renders correctly when there are no items', () => {
    const tree = renderer.create(<TextItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});