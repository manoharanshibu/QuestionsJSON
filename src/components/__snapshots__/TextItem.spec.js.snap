// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap shot test renders correctly when there are no items 1`] = `
<div
  class="node"
>
  <div
    className="tree-container"
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
</div>
`;
